### PR TITLE
fix: add remove history prompt, clear log area after remove

### PR DESCRIPTION
**Pull request by jakieoymj-github**
- [ ] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5345)

### CODE REVIEW

**Comment by jakieoymj-github** [comment]
Please forgive me. I am not very skilled with github.

File: src/model/about/aboutfriend.cpp [hunk 1]
```diff
@@ -105,8 +105,20 @@ bool AboutFriend::clearHistory()
     History* const history = Nexus::getProfile()->getHistory();
     if (history) {
         history->removeFriendHistory(pk.toString());
+        f->emitChatHistoryErasedSignal();
```
Comment (Diadlo) [inline src/model/about/aboutfriend.cpp]: 1. No one should manage signals of external class. I.e. it's not good practice to do `emit var.signal()` or `var.emitSignal()` where `emitSignal` just emit signal inside a `var`. Signal emitting should be result of some action or result of reaction on some action.
2. Also in current qTox architecture `Friend` is more like struct with information about Tox friend, than class with behaviour.

I'm not sure, what is the best way. But I have an idea. Add signal in `History` (something like `friendHistoryCleared(Friend*)`) and connect in `ChatForm` constructor. Don't forget to check `Friend` equals to friend stored in `ChatForm`.

**Review by Diadlo — COMMENTED** [review]
Glad to know, that I inspire someone :)

Yes, if you delete branch, github automatically closes your PR. But AFAIK there is button to restore it (somewhere near to ` @jakieoymj-github jakieoymj-github deleted the jakieoymj-github:j-fix-clearhistory branch`)

**Comment by jakieoymj-github** [comment]
Thanks a lot,  Actually I also have have considered this question. I will take time try to find a better way.

**Comment by jakieoymj-github** [comment]
@Diadlo Follow your advice, I do something like these to try to forward signal. Do not need any friendId or ToxPk.
1. Connect FriendWidget signal 'friendHistoryRemoved' with ChatForm signal 'clearChatArea' while FriendWidget object is instantiated.
2. Add a 'historyRemoved' signal in AboutFriend form, emit the signal when the 'Remove history (operation can not be undone!)' button clicked. Of course, we need to check whether history records are removed successful before emit the signal.
3. Connect AboutFriend form signal 'historyRemoved' with FriendWidget signal 'friendHistoryRemoved' while AboutFriend form object is instantiated.

**Comment by jakieoymj-github** [comment]
Oops, what's wrong? :(

**Comment by Diadlo** [comment]
@jakieoymj-github Looks like some flatpak error. Restarted

**Review by Diadlo — APPROVED** [review]
:lgtm_strong: from my side. Just a few style notes in comments

> Reviewed 7 of 14 files at r1, 8 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5345#-:-LNumdnR8txwjmZrIBNE:b-afgiu8)* status: :shipit: complete! 1 of 1 LGTMs obtained

___
*[src/persistence/history.h, line 76 at r2](https://reviewable.io/reviews/qtox/qtox/5345#-LNun1IZ24Ode5Gt0n67:-LNun1IZ24Ode5Gt0n68:byneinv) ([raw file](https://github.com/qtox/qtox/blob/5b6f59b2e02a8ea606c9cd73cc765c614735fbf0/src/persistence/history.h#L76)):*
> ```objc
>     void import(const HistoryKeeper& oldHistory);
> 
>     bool isHistoryExistence(const QString& friendPk);
> ```

Can you please add `// TODO: Make const`?
___
*[src/widget/about/aboutfriendform.cpp, line 111 at r2](https://reviewable.io/reviews/qtox/qtox/5345#-LNunFSC9ZKz3EB6UQ84:-LNunFSC9ZKz3EB6UQ85:bbxm4r7) ([raw file](https://github.com/qtox/qtox/blob/5b6f59b2e02a8ea606c9cd73cc765c614735fbf0/src/widget/about/aboutfriendform.cpp#L111)):*
> ```cpp
> void AboutFriendForm::onRemoveHistoryClicked()
> {
>     bool retYes = GUI::askQuestion(tr("Confirmation"),
> ```

I can be const `const bool retYes`
___
*[src/widget/about/aboutfriendform.cpp, line 118 at r2](https://reviewable.io/reviews/qtox/qtox/5345#-LNunID-AeDy5o4J_ofo:-LNunID-AeDy5o4J_ofp:bicahh3) ([raw file](https://github.com/qtox/qtox/blob/5b6f59b2e02a8ea606c9cd73cc765c614735fbf0/src/widget/about/aboutfriendform.cpp#L118)):*
> ```cpp
>     }
> 
>     bool rslt = about->clearHistory();
> ```

Come on. This variable used only once. `result` not so long to make it shorter :)
Also can be const

**Comment by jakieoymj-github** [comment]
@Diadlo  'Can you please add // TODO: Make const?' Do you mean just add // TODO: Make const and no change anything, then re-push the branch and re-PR?
OR:make change as follows, re-push and re-PR?
>const bool retYes = GUI::askQuestion(tr("Confirmation"),

>const bool result = about->clearHistory();

**Comment by Diadlo** [comment]
@jakieoymj-github Just amend your commit and force push

**Comment by jakieoymj-github** [comment]
I force push twice, it still was not successful. :6

**Comment by Diadlo** [comment]
@sudden6 looks like flatpak build is unstable. Was filed twice. Now passed successful

**Comment by sudden6** [comment]
@Diadlo what was the error?

**Comment by Diadlo** [comment]
@sudden6 Unfortunately didn't save an error. I think something like http 500 error

**Comment by sudden6** [comment]
I'll keep an eye on it.

**Pull request merged**